client/src/components/VisibilityControl/VisibilityControl.stories.js

```diff
@@ -0,0 +1,72 @@
+import { useArgs } from "@storybook/client-api";
+import _ from "lodash";
+import React from "react";
+
+import { VisibilityControl } from "./VisibilityControl";
+
+export default {
+  title: "Input/VisibilityControl",
+  component: VisibilityControl,
+
+  // Need decorators to use useArgs()
+  decorators: [(Story) => <div>{Story()}</div>],
+};
+
+const Template = (args) => {
+  const [_x, updateArgs] = useArgs();
+
+  const click_callback = (clicked_item_key) => {
+    console.log(
+      `${clicked_item_key} item has been clicked, consumer's callback fires.` +
+        "Example callback here toggles the clicked item, but actual behaviour is up to the consumer as this component itself is stateless"
+    );
+
+    updateArgs({
+      ...args,
+      items: _.map(args.items, ({ key, active, ...rest }) => ({
+        key,
+        active: key === clicked_item_key ? !active : active,
+        ...rest,
+      })),
+    });
+  };
+
+  return <VisibilityControl {...args} click_callback={click_callback} />;
+};
+
+const get_icon = (key) => (
```

Comment:
Stories, since they're able to act like visual tests for our components, should really only intersect with the source code at the point of the component they correspond to, IMO. Same reasons unit tests shouldn't depend on multiple source modules.

If we specifically write any integration stories, of course, that rule relaxes.